app_flutter/lib/task_box.dart

```diff
@@ -35,17 +38,31 @@ class TaskBox extends StatelessWidget {
     statusInProgress: Colors.blue,
     statusSkipped: Colors.transparent,
     statusSucceeded: Colors.green,
+    statusSucceededButFlaky: Colors.yellow,
     statusUnderperformed: Colors.orange,
+    statusUnderperformedInProgress: Colors.orange,
   };
 
   @override
   Widget build(BuildContext context) {
+    final bool attempted = task.attempts > 1;
+    if (attempted) {
+      if (task.status == statusSucceeded) {
+        task.status = statusSucceededButFlaky;
```

Comment:
I don't think you should be mutating the values of the task you are provided. Conceptually, some parent widget already "owns" this data, and might not be expecting it to change. Another way to think of this is that `build` methods should be side-effect free.

Instead I would keep some local that you update with the task status

Reply:
Updated to have a local status var. The original intention was to match how v1 was doing it, but I looked and v1 doesn't seem to advantage of mutating the status beyond what this widget does.